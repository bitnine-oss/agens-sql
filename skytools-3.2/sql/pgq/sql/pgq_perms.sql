\set ECHO off
\set VERBOSITY 'terse'
set client_min_messages = 'warning';

-- drop public perms
\i structure/newgrants_pgq.sql

-- select proname, proacl from pg_proc p, pg_namespace n where n.nspname = 'pgq' and p.pronamespace = n.oid;

\set ECHO all

drop role if exists pgq_test_producer;
drop role if exists pgq_test_consumer;
drop role if exists pgq_test_admin;

create role pgq_test_consumer with login in role pgq_reader;
create role pgq_test_producer with login in role pgq_writer;
create role pgq_test_admin with login in role pgq_admin;


\c - pgq_test_admin

select * from pgq.create_queue('pqueue'); -- ok

\c - pgq_test_producer

select * from pgq.create_queue('pqueue'); -- fail

select * from pgq.insert_event('pqueue', 'test', 'data'); -- ok

select * from pgq.register_consumer('pqueue', 'prod'); -- fail

\c - pgq_test_consumer

select * from pgq.create_queue('pqueue'); -- fail
select * from pgq.insert_event('pqueue', 'test', 'data'); -- fail
select * from pgq.register_consumer('pqueue', 'cons'); -- ok
select * from pgq.next_batch('pqueue', 'cons'); -- ok

