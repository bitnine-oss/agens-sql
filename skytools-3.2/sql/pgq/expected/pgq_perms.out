\set ECHO off
drop role if exists pgq_test_producer;
drop role if exists pgq_test_consumer;
drop role if exists pgq_test_admin;
create role pgq_test_consumer with login in role pgq_reader;
create role pgq_test_producer with login in role pgq_writer;
create role pgq_test_admin with login in role pgq_admin;
\c - pgq_test_admin
select * from pgq.create_queue('pqueue'); -- ok
 create_queue 
--------------
            1
(1 row)

\c - pgq_test_producer
select * from pgq.create_queue('pqueue'); -- fail
ERROR:  permission denied for function create_queue
select * from pgq.insert_event('pqueue', 'test', 'data'); -- ok
 insert_event 
--------------
            1
(1 row)

select * from pgq.register_consumer('pqueue', 'prod'); -- fail
ERROR:  permission denied for function register_consumer
\c - pgq_test_consumer
select * from pgq.create_queue('pqueue'); -- fail
ERROR:  permission denied for function create_queue
select * from pgq.insert_event('pqueue', 'test', 'data'); -- fail
ERROR:  permission denied for function insert_event
select * from pgq.register_consumer('pqueue', 'cons'); -- ok
 register_consumer 
-------------------
                 1
(1 row)

select * from pgq.next_batch('pqueue', 'cons'); -- ok
 next_batch 
------------
           
(1 row)

